src/services/db/RepoService.ts

```diff
@@ -144,6 +151,51 @@ export default class RepoService extends GitHubService {
     })
   }
 
+  async readMediaFile(
+    sessionData: UserWithSiteSessionData,
+    { fileName, directoryName }: { fileName: string; directoryName: string }
+  ): Promise<any> {
+    logger.debug(`Reading media file: ${fileName}`)
+    logger.debug(`Reading directoryName: ${directoryName}`)
+    const { siteName } = sessionData
+
+    // fetch from local disk
+    if (this.isRepoWhitelisted(siteName)) {
+      logger.info(
+        `Reading media file from disk. Sitname: ${siteName}, directory name: ${directoryName}, fileName: ${fileName},`
+      )
+      const result = await this.gitFileSystemService.readMediaFile(
+        siteName,
+        directoryName,
+        fileName
+      )
+      if (result.isErr()) {
+        throw result.error
+      }
+
+      return result.value
+    }
+
+    // fetch from Github
+    const directoryData = await super.readDirectory(sessionData, {
```

Comment:
Should we do a `try catch` to handle potential error from calling `super.readDirectory` here?

Reply:
I think this is okay because it's the same behaviour as existing - any errors get caught by our overall error handler, and we don't need to catch and convert the error we receive at this point to something else

Reply:
Thanks @alexanderleegs for the explanation, I will look more into our overall error handler logic. Will resolve these two comments then.